src/main/webapp/WEB-INF/static/js/trialmanager/environments.js

```diff
@@ -67,21 +70,90 @@
 						modalInstance.result.then(deferred.resolve);
 						modalInstance.result.then((isOK) => {
 							if (isOK) {
-								datasetService.removeVariables(studyContext.trialDatasetId, deleteVariables).then(() => {
-									$scope.nested.dataTable.rerender();
-								});
+								$scope.detachOrDeleteFilesIfAny(deleteVariables);
 							}
 						});
 					} else {
-						deferred.resolve(true);
-						datasetService.removeVariables(studyContext.trialDatasetId, deleteVariables).then(() => {
-							$scope.nested.dataTable.rerender();
-						});
+						$scope.detachOrDeleteFilesIfAny(deleteVariables).then(deferred.resolve(true));
 					}
 				});
 				return deferred.promise;
 			};
 
+			$scope.detachOrDeleteFilesIfAny = async function (variableIds) {
+				if ($scope.isFileStorageConfigured) {
+					const fileCountResp = await fileService.getFileCount(variableIds, studyContext.trialDatasetId, null);
+					const fileCount = parseInt(fileCountResp.headers('X-Total-Count'));
+
+					if (true) {
```

Comment:
Why is this always true?

Reply:
Hi @abatac , I accidentally pushed this change(for local testing purposes). The conditional should be (fileCount > 0)